### PR TITLE
Add option to assert on training and/or validation loss

### PR DESCRIPTION
**Pull request by taylanbil**
* No asserts in case of bf16
* defaults to no asserts
* --assert_valid_loss and --assert_train_loss take on float values that act as the value to compare against.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -485,6 +485,18 @@ def keep_training(lr, epoch_itr, trainers):
 
     train_meter.stop()
     print('| done training in {:.1f} seconds'.format(train_meter.sum))
+    assert_on_losses(args, trainers)
+
+
+def assert_on_losses(args, trainers):
+    if xu.getenv_as('XLA_USE_BF16', bool, False):
+        # XXX: loss values are meaningless in this case due to precision in bf16
```
Comment (jysohn23) [inline train.py:493]: Should we log a warning?

File: train.py [hunk 2]
```diff
@@ -543,6 +555,8 @@ def get_args():
     parser.add_argument('--num_cores', type=int, default=8)
     parser.add_argument('--metrics_debug', action='store_true')
     parser.add_argument('--use_gpu', action='store_true')
+    parser.add_argument('--assert_train_loss', type=float, default=None)
```
Comment (jysohn23) [inline train.py]: For sake of consistency can we make these `--target_train_loss` etc?

https://github.com/pytorch/xla/blob/master/test/args_parse.py#L28

**Review by zcain117 — APPROVED** [review]

File: train.py (hunk 1 above)
Comment (taylanbil) [reply train.py:493]: we already do in line 608

**Review by jysohn23 — APPROVED** [review]
